src/ui-kit/theme/ThemeProvider.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useMemo, useState } from 'react';
+import { ThemeProvider } from '@mui/material';
+import { createCommonTheme, palette } from './theme';
+
+export const ThemeContext = createContext();
+
+export const CombinedThemeProvider = ({ children }) => {
+  const [currentTheme, setCurrentTheme] = useState('light');
+
+  const toggleTheme = () => {
+    setCurrentTheme(currentTheme === 'light' ? 'dark' : 'light');
```

Comment:
It is better to use prevState.
The theme should be stored in localStorage so that you don't need to change the theme again after updating the pages